packages/router/src/bindings/contractReader/index.ts

```diff
@@ -72,21 +84,41 @@ export const handleActiveTransactions = async (transactions: ActiveTransaction<a
       transaction.crosschainTx.invariant.transactionId,
     );
     if (handlingTracker.has(transaction.crosschainTx.invariant.transactionId)) {
-      logger.info("Already handling transaction", requestContext, methodContext);
+      logger.debug("Already handling transaction", requestContext, methodContext);
       continue;
     }
-    handlingTracker.set(transaction.crosschainTx.invariant.transactionId, transaction.status);
-    handleSingle(transaction, requestContext).finally(() =>
-      handlingTracker.delete(transaction.crosschainTx.invariant.transactionId),
-    );
+
+    let chainId: number;
+    if (
+      transaction.status === CrosschainTransactionStatus.SenderPrepared ||
+      transaction.status === CrosschainTransactionStatus.SenderExpired
+    ) {
+      chainId = transaction.crosschainTx.invariant.receivingChainId;
+    } else {
+      chainId = transaction.crosschainTx.invariant.sendingChainId;
+    }
+
+    handlingTracker.set(transaction.crosschainTx.invariant.transactionId, {
+      blockNumber: 0,
```

Comment:
block number 0?

Reply:
Oh I see, you set the block number below

If you're going to set a "substitute" or fake initial block number first, can we make it -1 so it's obvious it's just not set yet?

Reply:
nit

Reply:
yeah we can do that